### ISSUE TITLE
Play services dependency issue

### ISSUE THREAD

**Issue by rigo-ptz** [OneSignal/OneSignal-Android-SDK#1470]
**Description:**

I tried to integrate the SDK by following the documentation https://documentation.onesignal.com/docs/android-sdk-setup#step-2-add-onesignal-gradle-plugin-and-sdk but Gradle sync fails. Could you check play-services-base version, please?

```
In project 'app' a resolved Google Play services library dependency depends on another at an exact version (e.g. "[17.0.
0, 17.6.99]", but isn't being resolved to that version. Behavior exhibited by the library will be unknown.

Dependency failing: com.onesignal:OneSignal:4.6.1 -> com.google.android.gms:play-services-base@[17.0.0, 17.6.99], but pl
ay-services-base version was 17.6.0.

...

For extended debugging info execute Gradle from the command line with ./gradlew --info :app:assembleDebug to see the dep
endency paths to the artifact. This error message came from the strict-version-matcher-plugin Gradle plugin, report issu
es at https://github.com/google/play-services-plugins and disable by removing the reference to the plugin ("apply 'stric
t-version-matcher-plugin'") from build.gradle.
```
**Environment**

1. classpath 'gradle.plugin.com.onesignal:onesignal-gradle-plugin:0.13.4'
2. 'com.onesignal:OneSignal:[4.0.0, 4.99.99]',

**Steps to Reproduce Issue:**

1. Add the following dependencies 
"com.google.android.gms:play-services-base:17.6.0",
                    "com.google.android.gms:play-services-maps:17.0.1",
                    "com.google.android.gms:play-services-auth:19.2.0,
                    "com.google.android.gms:play-services-wallet:18.1.3",
2. Add OneSignal dependency.
3. Sync Gradle

**Comment by rigo-ptz** [comment]
I checked and it looks like removing 'com.google.android.gms.strict-version-matcher-plugin' doesn't help.

**Comment by rigo-ptz** [comment]
Well, these changes helped to eliminate the problem:
1. Removing `classpath 'com.google.android.gms:strict-version-matcher-plugin:1.2.2'` from buildscript.dependencies
2. Removing `apply plugin: 'com.google.android.gms.strict-version-matcher-plugin'` from app-level gradle file.
3. Adding `googleServices { disableVersionCheck = true }` to app-level gradle file.

But is the third step actually the correct way? For me, it looks like a workaround.

(I also found a similar solution for a similar problem here https://github.com/OneSignal/OneSignal-Gradle-Plugin/issues/37)

**Comment by jkasten2** [comment]
@rigo-ptz Thanks for pointing that out! I believe the root issue is due to an incompatibility with `strict-version-matcher-plugin` and `onesignal-gradle-plugin`. Since `onesignal-gradle-plugin` will bump versions higher if it finds something incompatible with OneSignal. The version change might be done in a way that `strict-version-matcher-plugin` doesn't detect correctly, or maybe `strict-version-matcher-plugin` does not handle version ranges in correctly in some cases. 

I did find the exact line where this message is being printed from, it's possible this could be helpful in finding a solution to get them working together in the future:
https://github.com/google/play-services-plugins/blob/8609452db3f23989797eae3e2494aff80ffac311/strict-version-matcher-plugin/src/main/java/com/google/android/gms/dependencies/DependencyInspector.java#L296

**Comment by rigo-ptz** [comment]
@jkasten2 Thank you! I think you can close this issue. 

By the way, I found another way to reproduce this even without `strict-version-matcher-plugin`. I keep all analytical libs (firebase, appsflyer, onesignal ...) in a separate android module. But in order to be able to init these libraries from the application class, I add dependencies there using `api "foo.bar:0.0.0"`. When I add OneSignal, I get the same error ```
In project 'app' a resolved Google Play services library dependency depends on another at an exact version (e.g. "[17.0.
0, 17.6.99]", but isn't being resolved to that version. 
```
To fix this, I had to add `googleServices { disableVersionCheck = true } ` to the app-level gradle.

**Comment by jkasten2** [comment]
This has now been fixed in the 4.6.4 release when using Gradle 6 or newer.
https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/4.6.4
